src/coreclr/ToolBox/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -2489,21 +2416,20 @@ void MethodContext::recGetArgType(CORINFO_SIG_INFO*       sig,
                                   DWORD                   exceptionCode)
 {
     if (GetArgType == nullptr)
-        GetArgType = new LightWeightMap<GetArgTypeValue, Agnostic_GetArgType_Value>();
+        GetArgType = new LightWeightMap<Agnostic_GetArgType_Key, Agnostic_GetArgType_Value>();
 
-    GetArgTypeValue key;
+    Agnostic_GetArgType_Key key;
     ZeroMemory(&key, sizeof(key)); // We use the input structs as a key and use memcmp to compare.. so
                                    // we need to zero out padding too
-    // Only setting values for things the EE seems to pay attention to... this is necessary since some of the values
+
+    // Only setting values for CORINFO_SIG_INFO things the EE seems to pay attention to... this is necessary since some of the values
```

Comment:
Not your diff, but do you know what this means?

Reply:
The `getArgType` and `getArgClass` APIs take a CORINFO_SIG_INFO as input, but the original author of SuperPMI must have noticed that we don't actually use most of it. You'd think we should create and store an Agnostic_CORINFO_SIG_INFO but instead we only store a few specific fields from that in the `Agnostic_GetArgType_Key` and `Agnostic_GetArgClass_Key` types. I haven't investigated why that works, but presumably it's because the JIT either never sets the other fields, or the VM doesn't consult them for these API calls.